### PR TITLE
[WIP] Generate french container names

### PR DESCRIPTION
**Pull request by dgageot**
This is still work in progress and a first step towards the support of a french version of Docker, namely Marcel.
- Still need to add names to the `NameGenerator`. By using ministers names, I should be able to find women names.
- Need to plug this french generator in somehow

Another PR should follow to rename all docker actions to marcel actions: `docker kill` -> `marcel extermine`

We might also as @shykes mentioned move this code to a container and find a way to plugin it in

Signed-off-by: David Gageot david@gageot.net

### CODE REVIEW

File: pkg/namesgenerator/names-generator-fr.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		"theodebald",
+	}
+)
+
+// GetRandomFrenchName generates a random name from the list of adjectives and surnames in this package
+// formatted as "adjective_surname". For example 'focused_turing'. If retry is non-zero, a random
+// integer between 0 and 10 will be added to the end of the name, e.g `focused_turing3`
+func GetRandomFrenchName(retry int) string {
+	rnd := random.Rand
+	name := fmt.Sprintf("%s_%s", leftFR[rnd.Intn(len(left))], rightFR[rnd.Intn(len(right))])
```
Comment (thaJeztah) [inline pkg/namesgenerator/names-generator-fr.go]: s/name/nom/ ? :wink:

**Comment by AdrieanKhisbe** [comment]
_Nom mais Oh_
Ils sont où les Mendès, Clémenceau, Rocard, Robespierre, Condorcet, Turgot, Compte ? :)

**Comment by t0b1nux** [comment]
panic: runtime error: index out of range --> something very bad happened

**Comment by cyphar** [comment]
Not sure if serious. :confused:

**Comment by t0b1nux** [comment]
@moviruo, I will lunch and then add your corrections as a PR on top of @dgageot branch

**Comment by cyphar** [comment]
This PR has just reminded me that `names-generator.go` needs to be renamed (dashes are ugly).

**Comment by t0b1nux** [comment]
Does my PR correct everything @moviuro ?

**Comment by dgageot** [comment]
@t0b1nux @moviuro The thing is, I'm not sure french accents are accepted for container names. I wanted to check first.

**Comment by t0b1nux** [comment]
@dgageot I hope so, otherwise this thread is a nonsense

**Comment by dgageot** [comment]
@t0b1nux Right now it's not supported. Yet another challenge to overcome

**Comment by t0b1nux** [comment]
Well, that's more funny then ;)

**Comment by AdrieanKhisbe** [comment]
At least there is no way to upset Clemenceau by mistake:
cf https://fr.wikipedia.org/wiki/Georges_Clemenceau#cite_note-1
:book:  :trollface: :tophat:

**Comment by cyphar** [comment]
Do all systems we support have unicode support (I'm looking at you, Windows)?

**Comment by t0b1nux** [comment]
Sorry, but I don't have windows, and I guess most docker users doesn't have it as well. If docker is mainly targeted at Linux (even if Microsoft is doing things), unicode support shouldn't be a problem

**Comment by cyphar** [comment]
Actually, Docker (the client **and the daemon**) can run on Windows.

**Comment by t0b1nux** [comment]
@cyphar Yeah, but only modern versions of it, right ? And thoses versions supports Unicode (I hope so)

**Comment by t0b1nux** [comment]
@dgageot Have you worked on supporting unicode in docker name generator or have any advice on how to do it ?

**Comment by jfrazelle** [comment]
it would be cooler if this was like a "french mode only thing ;)"

**Comment by dgageot** [comment]
Hi @jfrazelle, how do you imagine the "mode français" would be activated?

**Comment by jfrazelle** [comment]
Idk daemon flag maybe :)

**Comment by thaJeztah** [comment]
daemon *drapeau* probably

**Comment by dgageot** [comment]
Let's go for a *drapeau* then!

**Comment by calavera** [comment]
Thanks. I think we should not move forward with this.

**Comment by dgageot** [comment]
@calavera Yes, that was really more a joke than anything!!!

**Pull request closed without merging**